api/v1/lib/resources.go

```diff
@@ -89,291 +54,36 @@ func (t ResourceErrorType) Generate(reason string) error {
 			msg = reason
 		}
 	}
-	return &ResourceError{errorType: t, reason: msg}
+	return &resourceError{errorType: t, reason: msg}
 }
 
-func (err *ResourceError) Type() ResourceErrorType { return err.errorType }
-func (err *ResourceError) Reason() string          { return err.reason }
-func (err *ResourceError) Resource() Resource      { return err.spec }
+func (err *resourceError) Type() resourceErrorType { return err.errorType }
```

Comment:
this is silly, either expose `resourceErrorType` as a public type (and also the consts) or else nix this func